sklearn/naive_bayes.py

```diff
@@ -622,18 +622,26 @@ def fit(self, X, y, sample_weight=None):
         self._update_class_log_prior(class_prior=class_prior)
         return self
 
-    # XXX The following is a stopgap measure; we need to set the dimensions
-    # of class_log_prior_ and feature_log_prob_ correctly.
     def _get_coef(self):
         return (self.feature_log_prob_[1:]
                 if len(self.classes_) == 2 else self.feature_log_prob_)
 
+    def _warn_coef(self):
+        warnings.warn('coef_ and intercept_ will be deprecated',
+                      DeprecationWarning)
+        return self._get_coef()
+
     def _get_intercept(self):
         return (self.class_log_prior_[1:]
                 if len(self.classes_) == 2 else self.class_log_prior_)
 
-    coef_ = property(_get_coef)
-    intercept_ = property(_get_intercept)
+    def _warn_intercept(self):
+        warnings.warn('coef_ and intercept_ will be deprecated',
+                      DeprecationWarning)
+        return self._get_intercept()
+
+    coef_ = property(_warn_coef)
+    intercept_ = property(_warn_intercept)
```

Comment:
Following the comment of @jnothman above, you can remove this line.
I would also define the `coef_` as a property

```python
@property
def coef_(self):
    raise warning ...
```